app/Http/Requests/StorePostRequest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class StorePostRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize(){
+
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules(){
+
+        return [
+            'name' => ['required','max:255',Rule::unique('posts', 'name')->ignore($this->post)],
```

Comment:
Not sure where `$this->post` comes from, probably from user input. I think you are looking for `$this->route('post')`, but this is redundant except when updating or destroying.